plugins/analytics-module-ga4/README.md

```diff
@@ -0,0 +1,221 @@
+# Analytics Module: Google Analytics 4
+
+This plugin provides an opinionated implementation of the Backstage Analytics
+API for Google Analytics 4. Once installed and configured, analytics events will
+be sent to GA as your users navigate and use your Backstage instance.
+
+This plugin contains no other functionality.
+
+## Installation
+
+1. Install the plugin package in your Backstage app:
+   `cd packages/app && yarn add @backstage/plugin-analytics-module-ga4`
+2. Wire up the API implementation to your App:
+
+```tsx
+// packages/app/src/apis.ts
+import {
+  analyticsApiRef,
+  configApiRef,
+  identityApiRef,
+} from '@backstage/core-plugin-api';
+import { GoogleAnalytics4 } from '@backstage/plugin-analytics-module-ga4';
+
+export const apis: AnyApiFactory[] = [
+  // Instantiate and register the GA Analytics API Implementation.
+  createApiFactory({
+    api: analyticsApiRef,
+    deps: { configApi: configApiRef, identityApi: identityApiRef },
+    factory: ({ configApi, identityApi }) =>
+      GoogleAnalytics4.fromConfig(configApi, {
+        identityApi,
+      }),
+  }),
+];
+```
+
+3. Configure the plugin in your `app-config.yaml`:
+
+The following is the minimum configuration required to start sending analytics
+events to GA. All that's needed is your GA4 measurement ID:
+
+```yaml
+# app-config.yaml
+app:
+  analytics:
+    ga4:
+      measurementId: G-0000000-0
+```
+
+4. Update CSP in your `app-config.yaml`:
+
+The following is the minimal content security policy required to load scripts from GA.
+
+```yaml
+backend:
+  csp:
+    connect-src: ["'self'", 'http:', 'https:']
+    # Add these two lines below
+    script-src: ["'self'", "'unsafe-eval'", 'https://www.google-analytics.com']
+    img-src: ["'self'", 'data:', 'https://www.google-analytics.com']
+```
+
+## Configuration
+
+In order to be able to analyze usage of your Backstage instance by plugin, we recommend configuring [a content grouping](#enabling-content-grouping).
+Additional dimensional data can be captured using custom dimensions, like this:
+
+1. First, [configure the custom dimension in GA] [configure-custom-dimension].
+   Be sure to set the Scope to `Event`, and name it `dimension1`.
+2. Then, add a mapping to your `app.analytics.ga4` configuration that instructs
+   the plugin to capture Plugin IDs on the custom dimension you just created.
+   It should look like this:
```

Comment:
hey @sramakr, thanks for the contribution. I'm going through setup instructions and found some undocumented stuff
 * Configuring new dimension requires `Event parameter` to be set
 * There is no yaml example after `It should look like this:`

Reply:
Hey @edbighead! Would you mind opening up an issue to track? Afraid it’ll get lost otherwise.